src/controller/buffer-controller.ts

```diff
@@ -394,20 +464,23 @@ class BufferController extends EventHandler {
     const targetBackBufferPosition = currentTime - Math.max(liveBackBufferLength, this._levelTargetDuration);
 
     for (let index = bufferTypes.length - 1; index >= 0; index--) {
-      const bufferType = bufferTypes[index], buffered = sourceBuffer[bufferType].buffered;
-
-      // when target buffer start exceeds actual buffer start
-      if (buffered.length > 0 && targetBackBufferPosition > buffered.start(0)) {
-        // remove buffer up until current time minus minimum back buffer length (removing buffer too close to current
-        // time will lead to playback freezing)
-        // credits for level target duration - https://github.com/videojs/http-streaming/blob/3132933b6aa99ddefab29c10447624efd6fd6e52/src/segment-loader.js#L91
-        this.removeBufferRange(bufferType, sourceBuffer[bufferType], 0, targetBackBufferPosition);
+      const bufferType = bufferTypes[index];
+      const sb = sourceBuffer[bufferType as keyof SourceBufferSet];
+      if (sb) {
+        const buffered = sb.buffered;
+        // when target buffer start exceeds actual buffer start
+        if (buffered.length > 0 && targetBackBufferPosition > buffered.start(0)) {
+          // remove buffer up until current time minus minimum back buffer length (removing buffer too close to current
+          // time will lead to playback freezing)
+          // credits for level target duration - https://github.com/videojs/http-streaming/blob/3132933b6aa99ddefab29c10447624efd6fd6e52/src/segment-loader.js#L91
+          this.removeBufferRange(bufferType, sb, 0, targetBackBufferPosition);
+        }
       }
     }
   }
 
-  onLevelUpdated ({ details }) {
-    if (details.fragments.length > 0) {
+  onLevelUpdated ({ details }: { details: any}) {
```

Comment:
fragment looks like it might be a bit complicated and best left for now, but I think we could type the rest
```suggestion
  onLevelUpdated ({ details }: { details: { totalduration: number, targetduration?: number, averagetargetduration?: number, live: boolean, fragments: any[] }) {
```